karabo_data/validation.py

```diff
@@ -30,8 +31,36 @@ def validate(self):
     def run_checks(self):
         self.problems = []
         self.check_indices()
+        self.check_trainids()
         return self.problems
 
+    def record(self, msg, **kwargs):
+        self.problems.append(dict(
+            msg=msg, file=self.filename, **kwargs
+        ))
+
+    def check_trainids(self):
+        ds_path = 'INDEX/trainId'
```

Comment:
now that I think about it, I believe that `INDEX/trainId` should not contain zeroes at all, the ones that can contain zeros in datasets are in `INSTRUMENTS/...` but they don't always contain this field. Here we can maybe simply check if there are zeroes at all.

Reply:
At least some of the older datasets have the train IDs written in chunks, so there are zeroes at the end. E.g. `/gpfs/exfel/exp/SPB/201730/p900008/raw/r0273/RAW-R0273-DA01-S00000.h5`. It seems like the newer files don't do this, but it's not clear when that changed or if it's a feature we can rely on. `karabo_data` is also built to handle trailing 0s there.